packages/bruno-app/src/components/Sidebar/Collections/Collection/CollectionItem/GenerateCodeItem/utils/snippet-generator.js

```
@@ -67,9 +68,14 @@ const generateSnippet = ({ language, item, collection, shouldInterpolate = false
       request.params = interpolateParams(request.params, variables);
     }
 
-    // Build HAR request
+    // Always interpolate the URL for HAR generation (HTTPSnippet requires a valid URL)
+    const rawUrl = request.url;
+    const interpolatedUrl = interpolateUrl({ url: rawUrl, variables });
```

Biome (2.4.4)

[error] This variable is used before its declaration.

(lint/correctness/noInvalidUseBeforeDeclaration)